### ISSUE TITLE
Add support for correlated subqueries

### ISSUE THREAD

**Issue by dboeke** [turbot/steampipe#261]
Query:

```sql
select
  name,
  (
    select
      count(*)
    from
      aws_availability_zone az
    where
      az.region_name = r.name
  ) as az_count
from
  aws_region r;
```
expected result:
```
+----------------+----------+
| name           | az_count |
+----------------+----------+
| ap-northeast-1 | 5        |
| ap-northeast-2 | 5        |
| ap-northeast-3 | 3        |
| ap-south-1     | 3        |
| ap-southeast-1 | 3        |
| ap-southeast-2 | 3        |
| ca-central-1   | 3        |
| eu-central-1   | 3        |
| eu-north-1     | 3        |
| eu-west-1      | 3        |
| eu-west-2      | 3        |
| eu-west-3      | 3        |
| sa-east-1      | 3        |
| us-east-1      | 15       |
| us-east-2      | 3        |
| us-west-1      | 2        |
| us-west-2      | 10       |
+----------------+----------+
```

actual result

```
+----------------+----------+
| name           | az_count |
+----------------+----------+
| eu-north-1     | 3        |
| af-south-1     | 0        |
| sa-east-1      | 0        |
| eu-central-1   | 0        |
| eu-west-1      | 0        |
| ap-northeast-3 | 0        |
| ap-northeast-1 | 0        |
| ap-northeast-2 | 0        |
| us-east-1      | 0        |
| ap-southeast-1 | 0        |
| eu-west-3      | 0        |
| eu-west-2      | 0        |
| us-east-2      | 0        |
| ap-south-1     | 0        |
| ap-southeast-2 | 0        |
| eu-south-1     | 0        |
| me-south-1     | 0        |
| ap-east-1      | 0        |
| ca-central-1   | 0        |
| us-west-1      | 0        |
| us-west-2      | 0        |
+----------------+----------+
```

**Comment by kaidaguerre** [comment]
@dboeke I believe this is fixed in 0.3.0, please could you verify

**Comment by dboeke** [comment]
Yes, confirmed it works correctly in 0.3.0

**Comment by dboeke** [comment]
Fixed in v0.3.0
